OctoKit/NotificationThread.swift

```diff
@@ -0,0 +1,336 @@
+import Foundation
+import RequestKit
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+// TODO add enum for type and reason
+
+// MARK: - Model
+
+open class NotificationThreadSubject: Codable {
```

Comment:
Maybe add this class as `Subject` to the `NotificationThread` `class`? This way we could avoid the awkward naming and make it more explicit that this type is only used here.